flink-java/src/main/java/org/apache/flink/api/java/io/RowCsvInputFormat.java

```diff
@@ -197,6 +197,14 @@ protected boolean parseRecord(Object[] holders, byte[] bytes, int offset, int nu
 			if (startPos < 0) {
 				throw new ParseException(String.format("Unexpected parser position for column %1$s of row '%2$s'",
 					field, new String(bytes, offset, numBytes)));
+			} else if (startPos == limit
+					&& field != fieldIncluded.length - 1
+					&& !FieldParser.endsWithDelimiter(bytes, startPos - 1, fieldDelimiter)) {
+				if (isLenient()) {
```

Comment:
Add a comment: "We are at the end of the record, but not all fields have been read and the end is not a field delimiter indicating an empty last field."